src/test/java/io/cdap/plugin/cloud/vision/transform/transformer/FullTextAnnotationsToRecordTransformerTest.java

```diff
@@ -0,0 +1,290 @@
+    }
+
+    List<StructuredRecord> languages = actual.get(ImageExtractorConstants.TextParagraph.DETECTED_LANGUAGES_FIELD_NAME);
+    Assert.assertNotNull(languages);
+    Assert.assertEquals(expected.getProperty().getDetectedLanguagesList().size(), languages.size());
+    for (int i = 0; i < expected.getProperty().getDetectedLanguagesList().size(); i++) {
+      TextAnnotation.DetectedLanguage expectedLanguage = expected.getProperty().getDetectedLanguages(i);
+      StructuredRecord actualLanguage = languages.get(i);
+      assertLanguageEquals(expectedLanguage, actualLanguage);
+    }
+    String expectedBreak = expected.getProperty().getDetectedBreak().getType().name();
+    Assert.assertEquals(expectedBreak, actual.get(ImageExtractorConstants.TextParagraph.DETECTED_BREAK_FIELD_NAME));
+  }
+
+  private void assertWordEquals(Word expected, StructuredRecord actual) {
+    Assert.assertEquals(expected.getConfidence(),
+                        actual.<Float>get(ImageExtractorConstants.TextWord.CONFIDENCE_FIELD_NAME),
+                        DELTA);
+
+    List<StructuredRecord> paragraphs = actual.get(ImageExtractorConstants.TextWord.SYMBOLS_FIELD_NAME);
+    Assert.assertNotNull(paragraphs);
+    Assert.assertEquals(expected.getSymbolsList().size(), paragraphs.size());
+    for (int i = 0; i < expected.getSymbolsList().size(); i++) {
+      Symbol expectedSymbol = expected.getSymbols(i);
+      StructuredRecord actualSymbol = paragraphs.get(i);
+      assertSymbolEquals(expectedSymbol, actualSymbol);
+    }
+
+    List<StructuredRecord> languages = actual.get(ImageExtractorConstants.TextWord.DETECTED_LANGUAGES_FIELD_NAME);
+    Assert.assertNotNull(languages);
+    Assert.assertEquals(expected.getProperty().getDetectedLanguagesList().size(), languages.size());
+    for (int i = 0; i < expected.getProperty().getDetectedLanguagesList().size(); i++) {
+      TextAnnotation.DetectedLanguage expectedLanguage = expected.getProperty().getDetectedLanguages(i);
+      StructuredRecord actualLanguage = languages.get(i);
+      assertLanguageEquals(expectedLanguage, actualLanguage);
+    }
+    String expectedBreak = expected.getProperty().getDetectedBreak().getType().name();
+    Assert.assertEquals(expectedBreak, actual.get(ImageExtractorConstants.TextWord.DETECTED_BREAK_FIELD_NAME));
+  }
+
+  private void assertSymbolEquals(Symbol expected, StructuredRecord actual) {
+    Assert.assertEquals(expected.getText(), actual.get(ImageExtractorConstants.TextSymbol.TEXT_FIELD_NAME));
+    Assert.assertEquals(expected.getConfidence(),
+                        actual.<Float>get(ImageExtractorConstants.TextSymbol.CONFIDENCE_FIELD_NAME),
+                        DELTA);
+
+    List<StructuredRecord> languages = actual.get(ImageExtractorConstants.TextSymbol.DETECTED_LANGUAGES_FIELD_NAME);
+    Assert.assertNotNull(languages);
+    Assert.assertEquals(expected.getProperty().getDetectedLanguagesList().size(), languages.size());
```

Comment:
you should try to extract variables more often to avoid long repeatable chained calls

Reply:
Fixed.